hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -337,21 +407,34 @@ private void handleDeleteEvent(OMDBUpdateEvent<String, Object> event,
 
   private void handleSizeRelatedTableDeleteEvent(
       OMDBUpdateEvent<String, Object> event, String countKey,
-      String unReplicatedSizeKey, String replicatedSizeKey,
+      String unReplicatedSizeKey, String replicatedSizeKey, String tableName,
       HashMap<String, Long> objectCountMap,
       HashMap<String, Long> unreplicatedSizeCountMap,
-      HashMap<String, Long> replicatedSizeCountMap) {
+      HashMap<String, Long> replicatedSizeCountMap) throws IOException {
+
+    Boolean isFileTable = tableName.equals(OPEN_FILE_TABLE);
+    Boolean isKeyTable = tableName.equals(OPEN_KEY_TABLE);
+    Boolean isDeletedDirTable = tableName.equals(DELETED_DIR_TABLE);
+
     if (event.getValue() instanceof OmKeyInfo) {
       // Handle DELETE for OpenKeyTable & OpenFileTable
       OmKeyInfo omKeyInfo = (OmKeyInfo) event.getValue();
       objectCountMap.computeIfPresent(countKey,
           (k, count) -> count > 0 ? count - 1L : 0L);
-      unreplicatedSizeCountMap.computeIfPresent(unReplicatedSizeKey,
-          (k, size) -> size > omKeyInfo.getDataSize() ?
-              size - omKeyInfo.getDataSize() : 0L);
-      replicatedSizeCountMap.computeIfPresent(replicatedSizeKey,
-          (k, size) -> size > omKeyInfo.getReplicatedSize() ?
-              size - omKeyInfo.getReplicatedSize() : 0L);
+      if (isFileTable || isKeyTable) {
```

Comment:
Try to reduce if else complexity as this method can grow in future for file and key tables also.